### ISSUE TITLE
Add the ability to remove stacktrace elements when using AbstractAssert.failWithMessage()

### ISSUE THREAD

**Issue by obecker** [joel-costigliola/assertj-core#1042]
#### Summary
It would be nice for custom assertion classes using the provided method `failWithMessage(String, Object...)` if the resulting stacktrace doesn't contain stacktrace elements from the custom assertion class. 

Built-in assertions achieve that by using `Failures.removeAssertJRelatedElementsFromStackTraceIfNeeded(Throwable)` (which calls `Throwables.removeAssertJRelatedElementsFromStackTrace(Throwable)`). However, this method only removes AssertJ related elements (containing `org.assert` in the classname).

I could of course copy and adapt that code for my own assertion class, however, adding the ability to call the existing method with an additional class, class name, or package name parameter would make much more sense to me. Moreover, the above-mentioned method `failWithMessage` could be extended to take also that additional parameter and to invoke `removeAssertJRelatedElementsFromStackTrace` in its implementation.

#### Example

```java
// taken from the failWithMessage(..) JavaDoc
public TolkienCharacterAssert hasName(String name) {
  // check that actual TolkienCharacter we want to make assertions on is not null.
  isNotNull();
    
  // check condition
  if (!actual.getName().equals(name)) {
    failWithMessage("Expected character's name to be <%s> but was <%s>", name, actual.getName());
  }
    
  // return the current assertion for method chaining
  return this;
}
```
will currently print
```
java.lang.AssertionError: Expected character's name to be <Faramir> but was <Boromir>

	at org.assertj.core.api.AbstractAssert.failWithMessage(AbstractAssert.java:116)
	at my.domain.assertj.TolkienCharacterAssert.hasName(TolkienCharacterAssert.java:27)
	at my.domain.TolkienTest.shouldDemonstrateIssue(TolkienTest.java:19)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
```

when (for example) used like this

```
    failWithMessage(TolkienCharacterAssert.class, 
        "Expected character's name to be <%s> but was <%s>", name, actual.getName());
```
it should drop the first two stacktrace elements, thus starting the stacktrace with `at my.domain.TolkienTest.shouldDemonstrateIssue(TolkienTest.java:19)`
(maybe using a class parameter is too restrictive, then using a string might the way to go)

#### Java 8 specific ?

No

Does this make sense to you? I could work on a pull request. The question is whether the classes `Failures` and `Throwables` must be fully compatible with previous versions (i.e. no signature changes of existing methods).

**Comment by obecker** [comment]
I just realized that `failWithMessage()` could internally access `this`, particularly `this.getClass()` and with this information could remove all stacktrace elements after and including the actual class name. In that case we don't have to change/extend the API of `failWithMessage()`.

**Comment by joel-costigliola** [comment]
@obecker I agree there is not much value exposing the internal of AssertJ or custom assertions classes, I'm happy if you can create a  PR for this (as longs as it does not break any API).

**Comment by obecker** [comment]
Apparently I missed the latest releases and looked at outdated code. My bad.
The requested improvement has been already implemented (issue #873).
I will close this issue.

**Comment by joel-costigliola** [comment]
I guess I should have known that, thanks for closing the issue.
